### ISSUE TITLE
Unable to reference predefined variables while setting a variable

### ISSUE THREAD

**Issue by eschwartz** [MicrosoftDocs/vsts-docs#5351]
I want set a variable that is derived from a predefined variable.

For example:
```yaml
variables:
  namespace: $[ format('PR-{0}', System.PullRequest.PullRequestNumber) ]
```

This fails with:
```
./azure-pipelines.yml (Line: 19, Col: 14): Unrecognized value: 'System'. Located at position 1 within expression: System.PullRequest.PullRequestNumber. For more help, refer to https://go.microsoft.com/fwlink/?linkid=842996
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: dd7e0bd3-1f7d-d7b6-cc72-5ef63c31b46a
* Version Independent ID: dae87abd-b73d-9120-bcdb-6097d4b40f2a
* Content: [Variables - Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#set-variables-in-pipeline)
* Content Source: [docs/pipelines/process/variables.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/pipelines/process/variables.md)
* Product: **devops**
* Technology: **devops-cicd**
* GitHub Login: @steved0x
* Microsoft Alias: **sdanie**

**Comment by WilliamAntonRohm** [comment]
Thanks for your question. It looks like you're working through an issue with your scenario or implementation, rather than an issue with the documentation. Here are a couple of options where you might consider asking your question: 

- [Azure DevOps on Stack Overflow](https://stackoverflow.com/questions/tagged/azure-devops) 
- [Azure DevOps Support Bot](https://azuredevopsvirtualagent.azurewebsites.net/)

**Comment by eschwartz** [comment]
@WilliamAntonRohm  -- 
Let me rephrase my issue: I have in fact verified that you _cannot_ reference _Predefined Variables_ within the _variables_ configuration. This would be a nice thing to call out in the docs, and it took me quite a while to work through this by trial and error.

**Comment by note-do** [comment]
Disagree. You can reference predefined variables, but only in the case when they have been initialized on the build. In your case System.PullRequest.PullRequestNumber will be available only with GitHub PR. In Azure DevOps PR you have only:
6 queue time variables used
system.pullRequest.pullRequestId : 90
system.pullRequest.sourceBranch : refs/heads/feature/azure-pipelines
system.pullRequest.targetBranch : refs/heads/feature/testCI
system.pullRequest.sourceCommitId : ***
system.pullRequest.sourceRepositoryUri : ***
system.pullRequest.pullRequestIteration : 98
If you run Pipiline without PR and brunch policy you will not have acceess to the system.pullRequest* variables
